rolepermissions/management/commands/sync_roles.py

```diff
@@ -0,0 +1,27 @@
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.contrib.auth.models import Group
+from django.contrib.auth import get_user_model
+from rolepermissions import roles
+
+UserModel = get_user_model()
+
+class Command(BaseCommand):
+    ROLEPERMISSIONS_MODULE = getattr(settings, 'ROLEPERMISSIONS_MODULE', 'roles.py')
+    help = "Synchronize auth Groups and Permissions with UserRoles defined in %s."%ROLEPERMISSIONS_MODULE
+
+    def handle(self, *args, **options):
+        # Sync auth.Group with current registered roles (leaving existing groups intact!)
+        for role in roles.RolesManager.get_roles() :
+            group, created = Group.objects.get_or_create(name=role.get_name())
+            if created:
+                self.stdout.write("Created Group: %s from Role: %s"%(group.name, role.get_name()))
+            # Sync auth.Permission with permissions for this role
+            role.get_default_true_permissions()
+
+        # Push any permission changes made to roles and remove any unregistered roles from all auth.Users
+        for user in UserModel.objects.all():
```

Comment:
This is dangerous. Id rather have this as an example script in the documentation. This is something you only want to run if you are absolutely sure about what the script is doing. It could go right after the docs for the command. What do you think?

Reply:
Sure.  I think you are just talking about the second part that syncs User permissions, right?  Creating a Group for each defined Role so they are immediately available in the Admin seems harmless?

Reply:
Yes, the first part is cool. Only worried about the second (from line 22 on).

Reply:
I pulled out the second part, made it an optional parameter:

    sync_roles  --reset_user_permissions

Willing to document this clearly.